components/apps_modal/interactive_dialog.tsx

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+import {
+    checkDialogElementForError,
+} from 'mattermost-redux/utils/integration_utils';
+import {AppCallResponse, AppField, AppForm, AppModalState} from 'mattermost-redux/types/apps';
+import {DialogElement as DialogElementProps} from 'mattermost-redux/types/integrations';
+
+import SpinnerButton from 'components/spinner_button';
+
+import EmojiMap from 'utils/emoji_map';
+import {localizeMessage} from 'utils/utils.jsx';
+
+import DialogElement from './dialog_element';
+import DialogIntroductionText from './dialog_introduction_text';
+
+export type Props = {
+    modal: AppModalState;
+    appID?: string;
+    url: string;
+    callbackId?: string;
+    elements: DialogElementProps[],
+    title: string;
+    introductionText?: string;
+    iconUrl?: string;
+    submitLabel?: string;
+    notifyOnCancel?: boolean;
+    state?: string;
+    onHide: () => void,
+    actions: {
+        submit: (dialog: {
+            values: {
+                [name: string]: string;
+            };
+        }) => Promise<{data: AppCallResponse}>;
+    },
+    emojiMap: EmojiMap;
+}
+
+type State = {
+    show: boolean;
+    values: {[name: string]: string};
+    error: string | null;
+    errors: {[name: string]: React.ReactNode};
+    submitting: boolean;
+}
+
+export default class InteractiveDialog extends React.PureComponent<Props, State> {
```

Comment:
Any chance we can reuse the interactive dialog component already existing?

Reply:
Update: Daniel and I discussed offline that we will keep the existing InteractiveDialog and DialogElement components the same to avoid regressions